pkg/cloud/services/ec2/subnets.go

```diff
@@ -325,9 +328,7 @@ func (s *Service) getSubnetTagParams(id string, public bool) infrav1.BuildParams
 	}
 
 	// Add tag needed for Service type=LoadBalancer
-	additionalTags = infrav1.Tags{
-		infrav1.NameKubernetesAWSCloudProviderPrefix + s.scope.Name(): string(infrav1.ResourceLifecycleShared),
-	}
+	additionalTags[infrav1.NameKubernetesAWSCloudProviderPrefix+s.scope.Name()] = string(infrav1.ResourceLifecycleShared)
```

Comment:
It might make sense to add a different parameter to the infrav1.BuildParams to handle this use case, since it appears in multiple places and would remove the need to do the nil checks.

Reply:
The Merge will fix this